src/Bicep.LangServer/Deploy/DeploymentHelper.cs

```diff
@@ -81,40 +90,65 @@ public class DeploymentHelper
                     Location = location,
                 };
 
-                string deployment = "bicep_deployment_" + DateTime.UtcNow.ToString("yyyyMMddHHmmss");
-
                 try
                 {
-                    var deploymentCreateOrUpdateOperation = await deploymentCollection.CreateOrUpdateAsync(WaitUntil.Completed, deployment, armDeploymentContent);
+                    var deploymentOperation = await deploymentCollection.CreateOrUpdateAsync(WaitUntil.Started, deploymentName, armDeploymentContent);
+
+                    if (deploymentOperation is null)
+                    {
+                        return new BicepDeploymentStartResponse(false, string.Format(LangServerResources.DeploymentFailedMessage, documentPath), null);
+                    }
+
+                    deploymentOperationsCache.CacheDeploymentOperation(deploymentId, deploymentOperation);
 
-                    return GetDeploymentResultMessage(deploymentCreateOrUpdateOperation, documentPath);
+                    var linkToDeploymentInAzurePortal = GetLinkToDeploymentInAzurePortal(portalUrl, Uri.EscapeDataString(id), deploymentName);
```

Comment:
Shouldn't Uri.EscapeDataString be in GetLinkToDeploymentInAzurePortal?

Reply:
Updated here:
https://github.com/Azure/bicep/commit/2ada9663d5fb2a933d313877a6a8d2f9b026b10c